test/tracker-component.js

```diff
@@ -0,0 +1,305 @@
+    });
+
+    it('should send view events', () => {
+        const userID = '__test__userID2';
+        const userName = '__test__userName2';
+        const tracker = Tracker({ user: { id: userID, name: userName } });
+        expect(appendChildCalls).to.equal(0);
+        tracker.view({
+            pageUrl: 'https://example.com/test2'
+        });
+        expect(imgMock.src).to.equal(
+            'https://www.paypal.com/targeting/track/view?data=eyJwYWdlVXJsIjoiaHR0cHM6Ly9leGFtcGxlLmNvbS90ZXN0MiIsInVzZXIiOnsiaWQiOiJfX3Rlc3RfX3VzZXJJRDIiLCJuYW1lIjoiX190ZXN0X191c2VyTmFtZTIifSwidHJhY2tpbmdUeXBlIjoidmlldyIsImNsaWVudElkIjoiYWJjeHl6MTIzIiwibWVyY2hhbnRJZCI6Inh5eixoaWosbG1ubyJ9'
+        );
+        expect(JSON.stringify(extractDataParam(imgMock.src))).to.equal(
+            JSON.stringify({
+                pageUrl:      'https://example.com/test2',
+                user:         { id: '__test__userID2', name: '__test__userName2' },
+                trackingType: 'view',
+                clientId:     'abcxyz123',
+                merchantId:   'xyz,hij,lmno'
+            })
+        );
+        expect(appendChildCalls).to.equal(1);
+    });
+
+    it('should send addToCart events', () => {
+        const userID = '__test__userID3';
+        const userName = '__test__userName3';
+        const tracker = Tracker({ user: { id: userID, name: userName } });
+        expect(appendChildCalls).to.equal(0);
+        tracker.addToCart({
+            cartId: '__test__cartId',
+            items:  [
+                {
+                    id:  '__test__productId',
+                    url: 'https://example.com/__test__productId'
+                }
+            ],
+            emailCampaignId: '__test__emailCampaignId',
+            price:           12345.67,
+            currencyCode:    'USD',
+            keywords:        [ '__test__' ]
+        });
+        expect(imgMock.src).to.equal(
+            'https://www.paypal.com/targeting/track/cartEvent?data=eyJjYXJ0SWQiOiJfX3Rlc3RfX2NhcnRJZCIsIml0ZW1zIjpbeyJpZCI6Il9fdGVzdF9fcHJvZHVjdElkIiwidXJsIjoiaHR0cHM6Ly9leGFtcGxlLmNvbS9fX3Rlc3RfX3Byb2R1Y3RJZCJ9XSwiZW1haWxDYW1wYWlnbklkIjoiX190ZXN0X19lbWFpbENhbXBhaWduSWQiLCJwcmljZSI6MTIzNDUuNjcsImN1cnJlbmN5Q29kZSI6IlVTRCIsImtleXdvcmRzIjpbIl9fdGVzdF9fIl0sImNhcnRFdmVudFR5cGUiOiJhZGRUb0NhcnQiLCJ1c2VyIjp7ImlkIjoiX190ZXN0X191c2VySUQzIiwibmFtZSI6Il9fdGVzdF9fdXNlck5hbWUzIn0sInRyYWNraW5nVHlwZSI6ImNhcnRFdmVudCIsImNsaWVudElkIjoiYWJjeHl6MTIzIiwibWVyY2hhbnRJZCI6Inh5eixoaWosbG1ubyJ9'
+        );
+        expect(JSON.stringify(extractDataParam(imgMock.src))).to.equal(
+            JSON.stringify({
+                cartId: '__test__cartId',
+                items:  [
+                    {
+                        id:  '__test__productId',
+                        url: 'https://example.com/__test__productId'
+                    }
+                ],
+                emailCampaignId: '__test__emailCampaignId',
+                price:           12345.67,
```

Comment:
@bryanjenningz One minor thing regarding numbers: do you want to accept floats here? (Typical problems with JavaScript floating point numbers). I'm not sure what kind of backend processing happens when it receives this data, but my initial reaction is that weird bugs could be encountered in the future.

The Payments SDK, for example, uses [strings](https://developer.paypal.com/docs/checkout/integrate/#5-capture-the-transaction) (e.g. `'12345.67'`), and another strategy for this is [zero decimal currency](https://stripe.com/docs/currencies#zero-decimal).

@bluepnume what convention do you guys think is best, cents as strings?

Probably worth deciding on this before merging.

Reply:
Fixed this, it uses '12345.67' format now